scripts/replicate_figure_2.py

```diff
@@ -0,0 +1,200 @@
+from __future__ import annotations
+
+import json
+import random
+import warnings
+
+import numpy as np
+import pandas as pd
+import plotly.graph_objects as go
+import torch
+from tqdm import tqdm
+from transformers import AutoConfig
+
+from obvs.lenses import ClassicLogitLens, PatchscopeLogitLens, TokenIdentity
+from obvs.patchscope import ModelLoader
+
+torch.cuda.empty_cache()
+
+model_names = {
+    "llamatiny": "TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T",
+    "llama": "meta-llama/Llama-2-13b-hf",
+    "gpt2": "gpt2",
+    "mamba": "MrGonao/delphi-mamba-100k",
+    "mistral": "mistralai/Mistral-7B-v0.1",
+    "gptj": "EleutherAI/gpt-j-6B",
+    "gemma": "google/gemma-2b",
+}
+
+# Construct the path to the file
+file_path = "data/processed/sentences.json"
+
+# Open and read the file
+with open(file_path) as file:
+    data = json.load(file)
+
+# Define the device and model name
+device = "gpu"
+model_name = "gpt2"
+# Load the model
+model = ModelLoader.load(model_names[model_name], device)
+# Load the model configuration
+config = AutoConfig.from_pretrained(model_names[model_name])
+# Get the number of layers
+num_layers = config.num_hidden_layers
+# Define the layers
+layers = list(range(0, num_layers))
+# Define the number of sentences to run
+test_amount = 100
+# Initialize the arrays to store the surprisal values
+surprisals_log_classic_ll = np.zeros((test_amount, num_layers))
+surprisals_log_patchscope_ll = np.zeros((test_amount, num_layers))
+surprisals_log_ti = np.zeros((test_amount, num_layers))
+warnings.filterwarnings("ignore")  # Ignore all warnings
+
+# Loop through the data
```

Comment:
In general, we should avoid commenting on **what** the code is doing. Instead, comments should be explaining **why** we're doing something.

This and the next few comments are repeats of what each line does. They add noise and make the code harder to read.

Reply:
Thanks, very helpful. Removed unnecessary comments